File: frontendapp/lib/models/room.dart
class Room {
  final int id;
  final int cinemaId;
  final String name;
  final int seatColumnMax;
  final int seatRowMax;

  Room({
    required this.id,
    required this.cinemaId,
    required this.name,
    required this.seatColumnMax,
    required this.seatRowMax,
  });

  factory Room.fromJson(Map<String, dynamic> json)=> Room(
    id: json['id'],
    cinemaId: json['cinemaId'],
    name: json['name'],
    seatColumnMax: json['seatcolumnmax'],
    seatRowMax: json['seatrowmax'],
  );
}

File: backend/bookingmovieticketapp/app/Models/Room.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Room extends Model
{
    use HasFactory;
    protected $table = 'rooms';

    // Khóa chính
    protected $primaryKey = 'id';
    public $incrementing = true;
    public $timestamps = false;
    protected $fillable = [
        'cinemaid',
        'name',
        'seatcolumnmax',
        'seatrowmax'
    ];

    public function cinema()
    {
        return $this->belongsTo(Cinema::class, 'cinemaid');
    }
    public function seats()
    {
        return $this->hasMany(Seat::class, 'roomid');
    }

    public function showtimes()
    {
        return $this->hasMany(Showtime::class, 'roomid');
    }
}
